### PR TITLE
Sync PMI with upstream and clean up odds and ends

### PR DESCRIPTION
**Pull request by jdinan**

### CODE REVIEW

**Comment by jdinan** [comment]
This PR might be easiest to review patch-by-patch.

**Comment by jdinan** [comment]
Ready for review.

File: shmem_pmi/simple_pmi.c [hunk 1]
```diff
@@ -8,15 +8,17 @@
 /*
  * This file implements the client-side of the PMI interface.
  *
- * Note that the PMI client code must not print error messages (except
+ * Note that the PMI client code must not print error messages (except
```
Comment (kseager) [inline shmem_pmi/simple_pmi.c]: would you mind removing the trailing whitespace in some of these files?

Comment (jdinan) [inline shmem_pmi/simple_pmi.c]: We're downstream on these files, so the less we edit them the better.  With the changes in this PR, there are a total of 3 edits, to comment out includes of files we don't have.

What if I add a README instead, documenting the upstream source and the update procedure?

Comment (kseager) [inline shmem_pmi/simple_pmi.c]: that's true. I'll take the README option instead :)

**Pull request merged**